### DISCUSSION TITLE
[OP Stack Deployment] <Transaction Failed By Using Custom Gas Token >

### DISCUSSION THREAD

**Discussion by DavidCoder88** [ethereum-optimism/developers#590] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Frist, I deployed my local L2 chain by this document shows:https://docs.optimism.io/builders/chain-operators/features/custom-gas-token
Secondly, I successfully deposited ERC20 tokens to the Admin account.
At last, when I tested an transaction by metamask wallet, it failed instantly，never show the button to confirm.

### Logs

WARN [09-16|16:07:47.448] Served eth_sendRawTransaction            conn=45.79.112.29:43070 reqid=7557112384642149 duration="391.821µs" err="insufficient funds for gas * price + value: balance 101001000000000000000000, tx cost 8435078591738081490270179247172673193047406793573238656486289333918437285924193678, overshot 8435078591738081490270179247172673193047406793573238656486188332918437285924193678"

**Comment by hash1go** [comment]
Hello!
Could you please add information about the version of each component, the configuration, and what kind of transactions you attempted on which chain?

**Reply by DavidCoder88** [reply]
**components version**
geth version
Geth
Version: 1.10.23-stable
Git Commit: d901d85377c2c2f05f09f423c7d739c0feecd90a
Architecture: amd64
Go Version: go1.18.5
Operating System: linux
GOPATH=
GOROOT=go



op-node version v1.7.7-f8143c8c-1717593043

op-batcher version v1.7.7-f8143c8c-1717593043

op-proposer version v1.7.7-f8143c8c-1717593043

**genesis.json:**

{
  "config": {
    "chainId": 2159998,
    "homesteadBlock": 0,
    "eip150Block": 0,
    "eip155Block": 0,
    "eip158Block": 0,
    "byzantiumBlock": 0,
    "constantinopleBlock": 0,
    "petersburgBlock": 0,
    "istanbulBlock": 0,
    "muirGlacierBlock": 0,
    "berlinBlock": 0,
    "londonBlock": 0,
    "arrowGlacierBlock": 0,
    "grayGlacierBlock": 0,
    "mergeNetsplitBlock": 0,
    "shanghaiTime": 0,
    "bedrockBlock": 0,
    "regolithTime": 0,
    "canyonTime": 0,
    "terminalTotalDifficulty": 0,
    "terminalTotalDifficultyPassed": true,
    "optimism": {
      "eip1559Elasticity": 6,
      "eip1559Denominator": 50,
      "eip1559DenominatorCanyon": 250
    }
  },
  "nonce": "0x0",
  "timestamp": "0x66e50260",
  "extraData": "0x424544524f434b",
  "gasLimit": "0x1c9c380",
  "difficulty": "0x0",
  "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "coinbase": "0x4200000000000000000000000000000000000011",
  "alloc": {
    "0x2df833A2d5eeE9579Ce0abC72d38a4205d9f0434":{
      "balance":"100000000000000000000000"
    },
    "0000000000000000000000000000000000000000": {
      "balance": "0x1"
    },
    "0000000000000000000000000000000000000001": {
      "balance": "0x1"
    },
    "0000000000000000000000000000000000000002": {
      "balance": "0x1"
    },
    "0000000000000000000000000000000000000003": {
      "balance": "0x1"
...
}

**config/rollup.json**
{
  "genesis": {
    "l1": {
      "hash": "0x42eeb16b247dfb71d2732895bd6554474c6e5b032f446a5f2e340852f06d88b7",
      "number": 6687595
    },
    "l2": {
      "hash": "0x0f93f5652b862dbff24415df8fcebefaf57e637b9d5272429927d2c4201ffcf9",
      "number": 0
    },
    "l2_time": 1726284384,
    "system_config": {
      "batcherAddr": "0x964b385d9a69cc4f3fe63267d2516a8d17f55f38",
      "overhead": "0x0000000000000000000000000000000000000000000000000000000000000000",
      "scalar": "0x00000000000000000000000000000000000000000000000000000000000f4240",
      "gasLimit": 30000000
    }
  },
  "block_time": 10,
  "max_sequencer_drift": 600,
  "seq_window_size": 3600,
  "channel_timeout": 300,
  "l1_chain_id": 11155111,
  "l2_chain_id": 2159998,
  "regolith_time": 0,
  "canyon_time": 0,
  "batch_inbox_address": "0xff00000000000000000000000000000000042069",
  "deposit_contract_address": "0xxxxx12e1153b550a155de19c8cc99fb39964xxx",
  "l1_system_config_address": "0xxxxx35721a93fe6d59e9a08f6e620ec1300axxxx",
  "protocol_versions_address": "0x0000000000000000000000000000000000000000",
  "da_challenge_contract_address": "0x0000000000000000000000000000000000000000"
}


**boot scripts infomation:**

ubuntu@ip-host:~/test-layer-scripts/op/sequencer-node$ cat start-op-geth.sh
echo "== Running op-geth script"
nohup ../build/geth --config ./config.toml --verbosity=4 --gcmode=archive --nat=none > ./log/geth.log &

ubuntu@ip-host:~/test-layer-scripts/op/sequencer-node$ more ./config.toml
[Eth]
NetworkId = 2159998
RollupDisableTxPoolGossip = true
SyncMode = "full"

[Node]
DataDir = "./datadir"
HTTPHost = "0.0.0.0"
HTTPPort = 8515
HTTPCors = ["*"]
HTTPVirtualHosts = ["*"]
HTTPModules = ["web3", "debug", "eth", "txpool", "net", "engine", "admin"]
WSHost = "0.0.0.0"
WSPort = 8516
WSOrigins = ["*"]
WSModules = ["debug", "eth", "txpool", "net", "engine"]
AuthVirtualHosts = ["*"]
AuthAddr = "0.0.0.0"
AuthPort = 8521
JWTSecret = "./jwt.txt"

[Node.P2P]
ListenAddr = ":8503"
NetRestrict = ["172.31.0.0/16"]

ubuntu@ip-host:~/test-layer-scripts/op/sequencer-node$ cat start-op-node.sh
echo "== Running op-node script"
nohup ../build/op-node \
--l2=http://localhost:8521 \
--l2.jwt-secret=./jwt.txt \
--sequencer.enabled \
--sequencer.l1-confs=5 \
--verifier.l1-confs=4 \
--rollup.config=../config/rollup.json \
--rpc.addr=0.0.0.0 --rpc.port=8517  --rpc.enable-admin \
--l1=$L1_RPC_URL --l1.rpckind=alchemy \
--syncmode=consensus-layer \
--p2p.sequencer.key=$GS_SEQUENCER_PRIVATE_KEY \
--p2p.listen.tcp=8505 \
--p2p.listen.udp=8505 > ./log/node.log &

ubuntu@ip-host:~/test-layer-scripts/op/sequencer-node$ cat start-op-batcher.sh
echo "== Running op-batcher script"
nohup  ../build/op-batcher \
--l2-eth-rpc=http://localhost:8515 \
--rollup-rpc=http://localhost:8517 \
--poll-interval=20s \
--sub-safety-margin=6 \
--num-confirmations=1 \
--safe-abort-nonce-too-low-count=3 \
--resubmission-timeout=30s \
--rpc.addr=0.0.0.0 \
--rpc.port=8518 \
--rpc.enable-admin \
--max-channel-duration=1500 \
--wait-node-sync=true \
--l1-eth-rpc=$L1_RPC_URL \
--private-key=$GS_BATCHER_PRIVATE_KEY  > ./log/batcher.log &
 
ubuntu@ip-host:~/test-layer-scripts/op/sequencer-node$ cat start-op-proposer.sh
echo "== Running op-proposer script"
nohup ../build/op-proposer \
--poll-interval=12s --rpc.port=8520 --rollup-rpc=http://localhost:8517 \
--l2oo-address=0xxxxx44cD4A9404b18FACCd1666B26d9158xxxx \
--private-key=$GS_PROPOSER_PRIVATE_KEY \
--l1-eth-rpc=$L1_RPC_URL > ./log/proposer.log &

**Reply by DavidCoder88** [reply]
I attempted on L2 which deployed by optimism. I want to use the ERC20 token which is from Sepolia as the custom gas token for L2, so it should be a normal transaction for L2

**Comment by DavidCoder88** [comment]
We tested transaction by code or metamask wallet, all got same result：insufficient funds for gas * price.
The balance is enough, and the amount value is only 1.

**Comment by Chomtana** [comment]
It sounds like incorrect gas scalar parameters are set on L1.

Do you have the deploy-config.json file that you used to deploy the chain and the address of the deployed L1 contracts?

**Reply by DavidCoder88** [reply]
deploy-config.json:

{
  "useCustomGasToken": true,
  "customGasTokenAddress": "0x4bff082a07c50724FEce17d9ecFC6dE1FFxxxxxx",

  "l1StartingBlockTag": "0x42eeb16b247dfb71d2732895bd6554474c6e5b032f446a5f2e340852f06d88b7",

  "l1ChainID": 11155111,
  "l2ChainID": 2159998,
  "l2BlockTime": 10,
  "l1BlockTime": 12,

  "maxSequencerDrift": 600,
  "sequencerWindowSize": 3600,
  "channelTimeout": 300,

  "p2pSequencerAddress": "0x0Ba1C831b97C843E331ddDBD8C56fEcC2b2287B0",
  "batchInboxAddress": "0xff00000000000000000000000000000000042069",
  "batchSenderAddress": "0x964B385D9a69cc4F3fE63267d2516a8D17f44FA8",

  "l2OutputOracleSubmissionInterval": 120,
  "l2OutputOracleStartingBlockNumber": 0,
  "l2OutputOracleStartingTimestamp": 1726284384,

  "l2OutputOracleProposer": "0x19D05B03116CF307fb68fb2727Ab13Aafb1C41cC",
  "l2OutputOracleChallenger": "0x2df833A2d5eeE9579Ce0abC72d38a4205d9f042E",

  "finalizationPeriodSeconds": 12,

  "proxyAdminOwner": "{ADMIN_ACCOUNT}",
  "baseFeeVaultRecipient": "{ADMIN_ACCOUNT}",
  "l1FeeVaultRecipient": "{ADMIN_ACCOUNT}",
  "sequencerFeeVaultRecipient": "{ADMIN_ACCOUNT}",
  "finalSystemOwner": "{ADMIN_ACCOUNT}",
  "superchainConfigGuardian": "{ADMIN_ACCOUNT}",

  "baseFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "l1FeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "sequencerFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "baseFeeVaultWithdrawalNetwork": 0,
  "l1FeeVaultWithdrawalNetwork": 0,
  "sequencerFeeVaultWithdrawalNetwork": 0,

  "gasPriceOracleOverhead": 0,
  "gasPriceOracleScalar": 1000000,

  "enableGovernance": true,
  "governanceTokenSymbol": "test",
  "governanceTokenName": "test",
  "governanceTokenOwner": "{ADMIN_ACCOUNT}",

  "l2GenesisBlockGasLimit": "0x1c9c380",
  "l2GenesisBlockBaseFeePerGas": "0x3b9aca00",
  "l2GenesisRegolithTimeOffset": "0x0",

  "eip1559Denominator": 50,
  "eip1559DenominatorCanyon": 250,
  "eip1559Elasticity": 6,

  "l2GenesisEcotoneTimeOffset": null,
  "l2GenesisDeltaTimeOffset": null,
  "l2GenesisCanyonTimeOffset": "0x0",

  "systemConfigStartBlock": 0,

  "requiredProtocolVersion": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "recommendedProtocolVersion": "0x0000000000000000000000000000000000000000000000000000000000000000",

  "faultGameAbsolutePrestate": "0x03c7ae758795765c6664a5d39bf63841c71ff191e9189522bad8ebff5d4eca98",
  "faultGameMaxDepth": 44,
  "faultGameClockExtension": 0,
  "faultGameMaxClockDuration": 600,
  "faultGameGenesisBlock": 0,
  "faultGameGenesisOutputRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "faultGameSplitDepth": 14,
  "faultGameWithdrawalDelay": 604800,

  "preimageOracleMinProposalSize": 1800000,
  "preimageOracleChallengePeriod": 86400
}

**Reply by DavidCoder88** [reply]
ubuntu@ip-172-31-38-175:~/op/optimism/packages/contracts-bedrock/deployments$ more 11155111-deploy.json
{
  "AddressManager": "0x5b678Ca4771d266DF670012C9fd0A201A4197E09",
  "AnchorStateRegistry": "0x49DaBc4A5DA8b55096E4808362Ce440898c7fFc3",
  "AnchorStateRegistryProxy": "0x323cfAae51E8E658386a2d450515311807688b97",
  "DelayedWETH": "0x3C00220aA849CEBCD2b4c4b5f0f4B068532E81D2",
  "DelayedWETHProxy": "0x88e8dc5928a7A5C220AEd2ecc59bBB9c295a22b8",
  "DisputeGameFactory": "0xB8a6f14a3F4384d56E167009C92EE26a8e1Eb795",
  "DisputeGameFactoryProxy": "0xAE519c1E5a4B3768E78e71Bf9F5DaB6fbA0C551b",
  "L1CrossDomainMessenger": "0x32749705784CEd6560B8f552Cf28daEC5B9e7a25",
  "L1CrossDomainMessengerProxy": "0x55cEbC9e88eaEC06067884b37B937CE6Db78D5Ea",
  "L1ERC721Bridge": "0x0804250C7Bf8c59d6283ED51D4B5F36c2Fa75af5",
  "L1ERC721BridgeProxy": "0x8c0D25767dDb2969C42BEBc56ff5d698A508713e",
  "L1StandardBridge": "0xd1372f789F0E043Bf9c13ba6227c0Bc65854a7Cb",
  "L1StandardBridgeProxy": "0x2380703Aa5F782692259dD494259Cf37760DB028",
  "L2OutputOracle": "0x7C8B4A4A8DDfF81447e759380E48e27C47c03C82",
  "L2OutputOracleProxy": "0xd6ef0D1e0FBfbDcb422De45639D1ddC8cc9eAe1A",
  "Mips": "0xfe773d8e88E91C4B82Fca92568884A8B7aaB5C04",
  "OptimismMintableERC20Factory": "0xdCF4e5a556A8E2d9ae8E665AC92548EAC7D25c94",
  "OptimismMintableERC20FactoryProxy": "0x7FcAEfBA1cDC42BefF7ad0066c24cddf4b97824c",
  "OptimismPortal": "0x88378743216d8862F132432662f198B90f997E61",
  "OptimismPortal2": "0x6C35D315D54Ce15118FCF65fEbf634cE48B0be46",
  "OptimismPortalProxy": "0x316712E1153b550A155de19c8cc99fb3996446C8",
  "PreimageOracle": "0x95c9acFac640EFaffdDDF8335e137b7D77Edd33F",
  "ProtocolVersions": "0xCa28b17E069e642C4487BDb74fF6ccf5a67Ee0C1",
  "ProtocolVersionsProxy": "0x280cB714aA04a5B44bC42Cec8b92C73a37035370",
  "ProxyAdmin": "0x7CC5302FF0f02A355642a02abC88A05435937B84",
  "SafeProxyFactory": "0xa6B71E26C5e0845f74c812102Ca7114b6a896AB2",
  "SafeSingleton": "0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552",
  "SuperchainConfig": "0xbd52B27eA7436f281111EF480345Ff6dE1032a6d",
  "SuperchainConfigProxy": "0x5D2EC86CF7B8967529717C7389CBb33d35542E2E",
  "SystemConfig": "0x70AdB4a43701297576980bBCB4d331cCaA136222",
  "SystemConfigProxy": "0x693735721A93fe6d59E9a08f6e620ec1300A3a4A",
  "SystemOwnerSafe": "0xdCAf627f7Afe192914a091e822B3504A38541cfc"
}

**Reply by Chomtana** [reply]
![image](https://github.com/user-attachments/assets/02b925a3-afbc-48e7-816b-0b3b7364edc4)

scalar parameter in SystemConfigProxy [0x693735721A93fe6d59E9a08f6e620ec1300A3a4A](https://sepolia.etherscan.io/address/0x693735721A93fe6d59E9a08f6e620ec1300A3a4A#readProxyContract) is very weird...

**Reply by Chomtana** [reply, marked as answer]
Can you try calling `setGasConfig` function in SystemConfigProxy: https://sepolia.etherscan.io/address/0x693735721A93fe6d59E9a08f6e620ec1300A3a4A#writeProxyContract

![image](https://github.com/user-attachments/assets/fd8acffb-7e9c-4814-b9c8-15b536abd52a)

* _overhead: 0
* _scalar: 1000000

And wait for a few minutes and try again to see if infinite gas price problem is solved or not

**Reply by DavidCoder88** [reply]
Thanks a lot. I solved my problem now. 
**Is the real reason that these two parameters which are in deploy-config.json didn't take effect ？**

gasPriceOracleOverhead": 0.
gasPriceOracleScalar":1000000.


I find the description from docs as following:
(DEPRECATED) gasPriceOracleScalar
GasPriceOracleScalar represents the initial value of the gas scalar in the GasPriceOracle predeploy. Deprecated: Since Ecotone, this field is superseded by GasPriceOracleBaseFeeScalar and GasPriceOracleBlobBaseFeeScalar.

(DEPRECATED) gasPriceOracleOverhead
GasPriceOracleOverhead represents the initial value of the gas overhead in the GasPriceOracle predeploy. Deprecated: Since Ecotone, this field is superseded by GasPriceOracleBaseFeeScalar and GasPriceOracleBlobBaseFeeScalar.
